Penteract/Assets/Scripts/RangerProjectileScript.cpp

```diff
@@ -0,0 +1,48 @@
+#include "RangerProjectileScript.h"
+
+#include "GameplaySystems.h"
+#include "PlayerController.h"
+#include "Components/ComponentTransform.h"
+
+EXPOSE_MEMBERS(RangerProjectileScript) {
+	MEMBER(MemberType::INT, speed),
+		MEMBER(MemberType::FLOAT, life)
+};
+
+GENERATE_BODY_IMPL(RangerProjectileScript);
+
+void RangerProjectileScript::Start() {
+
+}
+
+void RangerProjectileScript::Update() {
+	if (life >= 0) {
+		life -= Time::GetDeltaTime();
+		ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
+		float3 aux = transform->GetGlobalRotation() * float3(0.0f, 0.0f, 1.0f);
+
+		if (itsVertical) {
+			if (!firstTime) {
+				transform->SetGlobalRotation(Quat::RotateAxisAngle(aux, (pi / 2)).Mul(transform->GetGlobalRotation()));
+				firstTime = true;
+			}
+		}
+
+		aux *= speed * Time::GetDeltaTime();
+		float3 newPosition = transform->GetGlobalPosition();
+		newPosition += aux;
+		transform->SetGlobalPosition(newPosition);
+	} else {
+		GameplaySystems::DestroyGameObject(&GetOwner());
+	}
+}
+
+void RangerProjectileScript::OnCollision(const GameObject& collidedWith) {
+	GameObject* parent = collidedWith.GetParent();
+	if (parent != nullptr) {
```

Comment:
This should be controlled on the OnCollision of the player. The OnCollision of the bullets, the only thing they should do is kill the bullet itself

Reply:
Copy that, changing it and on hold until player has collider